12-kubernetes-cluster.tf

```
@@ -125,10 +125,24 @@ resource "azurerm_kubernetes_cluster" "kubernetes_cluster" {
       condition     = var.enable_automatic_channel_upgrade_patch != true || can(regex("^1\\.\\d\\d$", var.kubernetes_cluster_version))
       error_message = "When automatic upgrades are enabled, kubernetes_cluster_version must only include major and minor versions, not the patch version e.g. 1.18 or 1.25"
     }
+    precondition {
+      condition     = var.node_os_maintenance_window_duration != null && var.node_os_maintenance_window_duration > 3
```

```suggestion
      condition     = var.node_os_maintenance_window_duration != null && var.node_os_maintenance_window_duration >= 4
```

Haven't read into what this value can be, is it possible to have non-integer? Will need to change if so

---

In the docs relating to this they have it as an integer and whole numbers in az cli and json config files, can run a test to see if it can accept string or decimal.

---

Easier to define it as >=4 though right? Easier to read quickly in the code too

---

yeah I see what you mean will make that change too